// @angular/material/slider/slider.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from '@angular/material/slider/slider';
import * as import2 from '@angular/forms/src/directives';
import * as import3 from '@angular/forms/src/form_providers';
import * as import4 from '@angular/forms/src/directives/radio_control_value_accessor';
import * as import5 from '@angular/material/core/gestures/MdGestureConfig';
import * as import7 from '@angular/platform-browser/src/dom/events/hammer_gestures';
import * as import9 from '@angular/core/src/linker/view';
import * as import10 from '@angular/core/src/linker/element';
import * as import11 from '@angular/core/src/linker/view_utils';
import * as import12 from '@angular/core/src/linker/view_type';
import * as import13 from '@angular/core/src/change_detection/change_detection';
import * as import14 from '@angular/core/src/linker/element_ref';
import * as import15 from '@angular/forms/src/directives/control_value_accessor';
import * as import16 from '@angular/core/src/metadata/view';
import * as import17 from '@angular/core/src/linker/component_factory';
import * as import18 from './slider.css';
var MdSliderModuleInjector = (function (_super) {
    __extends(MdSliderModuleInjector, _super);
    function MdSliderModuleInjector(parent) {
        _super.call(this, parent, [], []);
    }
    Object.defineProperty(MdSliderModuleInjector.prototype, "_RadioControlRegistry_3", {
        get: function () {
            if ((this.__RadioControlRegistry_3 == null)) {
                (this.__RadioControlRegistry_3 = new import4.RadioControlRegistry());
            }
            return this.__RadioControlRegistry_3;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(MdSliderModuleInjector.prototype, "_HAMMER_GESTURE_CONFIG_4", {
        get: function () {
            if ((this.__HAMMER_GESTURE_CONFIG_4 == null)) {
                (this.__HAMMER_GESTURE_CONFIG_4 = new import5.MdGestureConfig());
            }
            return this.__HAMMER_GESTURE_CONFIG_4;
        },
        enumerable: true,
        configurable: true
    });
    MdSliderModuleInjector.prototype.createInternal = function () {
        this._InternalFormsSharedModule_0 = new import2.InternalFormsSharedModule();
        this._FormsModule_1 = new import3.FormsModule();
        this._MdSliderModule_2 = new import1.MdSliderModule();
        return this._MdSliderModule_2;
    };
    MdSliderModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import2.InternalFormsSharedModule)) {
            return this._InternalFormsSharedModule_0;
        }
        if ((token === import3.FormsModule)) {
            return this._FormsModule_1;
        }
        if ((token === import1.MdSliderModule)) {
            return this._MdSliderModule_2;
        }
        if ((token === import4.RadioControlRegistry)) {
            return this._RadioControlRegistry_3;
        }
        if ((token === import7.HAMMER_GESTURE_CONFIG)) {
            return this._HAMMER_GESTURE_CONFIG_4;
        }
        return notFoundResult;
    };
    MdSliderModuleInjector.prototype.destroyInternal = function () {
    };
    return MdSliderModuleInjector;
}(import0.NgModuleInjector));
export var MdSliderModuleNgFactory = new import0.NgModuleFactory(MdSliderModuleInjector, import1.MdSliderModule);
var renderType_MdSlider_Host = null;
var _View_MdSlider_Host0 = (function (_super) {
    __extends(_View_MdSlider_Host0, _super);
    function _View_MdSlider_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_MdSlider_Host0, renderType_MdSlider_Host, import12.ViewType.HOST, viewUtils, parentInjector, declarationEl, import13.ChangeDetectorStatus.CheckAlways);
    }
    Object.defineProperty(_View_MdSlider_Host0.prototype, "_NG_VALUE_ACCESSOR_0_5", {
        get: function () {
            if ((this.__NG_VALUE_ACCESSOR_0_5 == null)) {
                (this.__NG_VALUE_ACCESSOR_0_5 = [this._MdSlider_0_4]);
            }
            return this.__NG_VALUE_ACCESSOR_0_5;
        },
        enumerable: true,
        configurable: true
    });
    _View_MdSlider_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('md-slider', rootSelector, null);
        this.renderer.setElementAttribute(this._el_0, 'tabindex', '0');
        this._appEl_0 = new import10.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_MdSlider0(this.viewUtils, this.injector(0), this._appEl_0);
        this._MdSlider_0_4 = new import1.MdSlider(new import14.ElementRef(this._el_0));
        this._appEl_0.initComponent(this._MdSlider_0_4, [], compView_0);
        compView_0.create(this._MdSlider_0_4, this.projectableNodes, null);
        var disposable_0 = this.renderer.listen(this._el_0, 'click', this.eventHandler(this._handle_click_0_0.bind(this)));
        var disposable_1 = this.renderer.listen(this._el_0, 'slide', this.eventHandler(this._handle_slide_0_1.bind(this)));
        var disposable_2 = this.renderer.listen(this._el_0, 'slidestart', this.eventHandler(this._handle_slidestart_0_2.bind(this)));
        var disposable_3 = this.renderer.listen(this._el_0, 'slideend', this.eventHandler(this._handle_slideend_0_3.bind(this)));
        var disposable_4 = this.renderer.listenGlobal('window', 'resize', this.eventHandler(this._handle_resize_0_4.bind(this)));
        var disposable_5 = this.renderer.listen(this._el_0, 'blur', this.eventHandler(this._handle_blur_0_5.bind(this)));
        this._expr_6 = import13.UNINITIALIZED;
        this._expr_7 = import13.UNINITIALIZED;
        this._expr_8 = import13.UNINITIALIZED;
        this._expr_9 = import13.UNINITIALIZED;
        this._expr_10 = import13.UNINITIALIZED;
        this.init([].concat([this._el_0]), [this._el_0], [
            disposable_0,
            disposable_1,
            disposable_2,
            disposable_3,
            disposable_4,
            disposable_5
        ], []);
        return this._appEl_0;
    };
    _View_MdSlider_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import1.MdSlider) && (0 === requestNodeIndex))) {
            return this._MdSlider_0_4;
        }
        if (((token === import15.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) {
            return this._NG_VALUE_ACCESSOR_0_5;
        }
        return notFoundResult;
    };
    _View_MdSlider_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if ((this.numberOfChecks === 0)) {
                this._MdSlider_0_4.ngAfterContentInit();
            }
        }
        var currVal_6 = this._MdSlider_0_4.disabled;
        if (import11.checkBinding(throwOnChange, this._expr_6, currVal_6)) {
            this.renderer.setElementClass(this._el_0, 'md-slider-disabled', currVal_6);
            this._expr_6 = currVal_6;
        }
        var currVal_7 = this._MdSlider_0_4.disabled;
        if (import11.checkBinding(throwOnChange, this._expr_7, currVal_7)) {
            this.renderer.setElementAttribute(this._el_0, 'aria-disabled', ((currVal_7 == null) ? null : currVal_7.toString()));
            this._expr_7 = currVal_7;
        }
        var currVal_8 = this._MdSlider_0_4.min;
        if (import11.checkBinding(throwOnChange, this._expr_8, currVal_8)) {
            this.renderer.setElementAttribute(this._el_0, 'aria-valuemin', ((currVal_8 == null) ? null : currVal_8.toString()));
            this._expr_8 = currVal_8;
        }
        var currVal_9 = this._MdSlider_0_4.max;
        if (import11.checkBinding(throwOnChange, this._expr_9, currVal_9)) {
            this.renderer.setElementAttribute(this._el_0, 'aria-valuemax', ((currVal_9 == null) ? null : currVal_9.toString()));
            this._expr_9 = currVal_9;
        }
        var currVal_10 = this._MdSlider_0_4.value;
        if (import11.checkBinding(throwOnChange, this._expr_10, currVal_10)) {
            this.renderer.setElementAttribute(this._el_0, 'aria-valuenow', ((currVal_10 == null) ? null : currVal_10.toString()));
            this._expr_10 = currVal_10;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_MdSlider_Host0.prototype._handle_click_0_0 = function ($event) {
        this._appEl_0.componentView.markPathToRootAsCheckOnce();
        var pd_0 = (this._MdSlider_0_4.onClick($event) !== false);
        return (true && pd_0);
    };
    _View_MdSlider_Host0.prototype._handle_slide_0_1 = function ($event) {
        this._appEl_0.componentView.markPathToRootAsCheckOnce();
        var pd_0 = (this._MdSlider_0_4.onSlide($event) !== false);
        return (true && pd_0);
    };
    _View_MdSlider_Host0.prototype._handle_slidestart_0_2 = function ($event) {
        this._appEl_0.componentView.markPathToRootAsCheckOnce();
        var pd_0 = (this._MdSlider_0_4.onSlideStart($event) !== false);
        return (true && pd_0);
    };
    _View_MdSlider_Host0.prototype._handle_slideend_0_3 = function ($event) {
        this._appEl_0.componentView.markPathToRootAsCheckOnce();
        var pd_0 = (this._MdSlider_0_4.onSlideEnd() !== false);
        return (true && pd_0);
    };
    _View_MdSlider_Host0.prototype._handle_resize_0_4 = function ($event) {
        this._appEl_0.componentView.markPathToRootAsCheckOnce();
        var pd_0 = (this._MdSlider_0_4.onResize() !== false);
        return (true && pd_0);
    };
    _View_MdSlider_Host0.prototype._handle_blur_0_5 = function ($event) {
        this._appEl_0.componentView.markPathToRootAsCheckOnce();
        var pd_0 = (this._MdSlider_0_4.onBlur() !== false);
        return (true && pd_0);
    };
    return _View_MdSlider_Host0;
}(import9.AppView));
function viewFactory_MdSlider_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_MdSlider_Host === null)) {
        (renderType_MdSlider_Host = viewUtils.createRenderComponentType('', 0, import16.ViewEncapsulation.None, [], {}));
    }
    return new _View_MdSlider_Host0(viewUtils, parentInjector, declarationEl);
}
export var MdSliderNgFactory = new import17.ComponentFactory('md-slider', viewFactory_MdSlider_Host0, import1.MdSlider);
var styles_MdSlider = [import18.styles];
var renderType_MdSlider = null;
var _View_MdSlider0 = (function (_super) {
    __extends(_View_MdSlider0, _super);
    function _View_MdSlider0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_MdSlider0, renderType_MdSlider, import12.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import13.ChangeDetectorStatus.CheckAlways);
    }
    _View_MdSlider0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'md-slider-wrapper');
        this._text_1 = this.renderer.createText(this._el_0, '\n  ', null);
        this._el_2 = this.renderer.createElement(this._el_0, 'div', null);
        this.renderer.setElementAttribute(this._el_2, 'class', 'md-slider-container');
        this._text_3 = this.renderer.createText(this._el_2, '\n    ', null);
        this._el_4 = this.renderer.createElement(this._el_2, 'div', null);
        this.renderer.setElementAttribute(this._el_4, 'class', 'md-slider-track-container');
        this._text_5 = this.renderer.createText(this._el_4, '\n      ', null);
        this._el_6 = this.renderer.createElement(this._el_4, 'div', null);
        this.renderer.setElementAttribute(this._el_6, 'class', 'md-slider-track');
        this._text_7 = this.renderer.createText(this._el_4, '\n      ', null);
        this._el_8 = this.renderer.createElement(this._el_4, 'div', null);
        this.renderer.setElementAttribute(this._el_8, 'class', 'md-slider-track md-slider-track-fill');
        this._text_9 = this.renderer.createText(this._el_4, '\n      ', null);
        this._el_10 = this.renderer.createElement(this._el_4, 'div', null);
        this.renderer.setElementAttribute(this._el_10, 'class', 'md-slider-tick-container');
        this._text_11 = this.renderer.createText(this._el_4, '\n      ', null);
        this._el_12 = this.renderer.createElement(this._el_4, 'div', null);
        this.renderer.setElementAttribute(this._el_12, 'class', 'md-slider-last-tick-container');
        this._text_13 = this.renderer.createText(this._el_4, '\n    ', null);
        this._text_14 = this.renderer.createText(this._el_2, '\n    ', null);
        this._el_15 = this.renderer.createElement(this._el_2, 'div', null);
        this.renderer.setElementAttribute(this._el_15, 'class', 'md-slider-thumb-container');
        this._text_16 = this.renderer.createText(this._el_15, '\n      ', null);
        this._el_17 = this.renderer.createElement(this._el_15, 'div', null);
        this.renderer.setElementAttribute(this._el_17, 'class', 'md-slider-thumb-position');
        this._text_18 = this.renderer.createText(this._el_17, '\n        ', null);
        this._el_19 = this.renderer.createElement(this._el_17, 'div', null);
        this.renderer.setElementAttribute(this._el_19, 'class', 'md-slider-thumb');
        this._text_20 = this.renderer.createText(this._el_17, '\n        ', null);
        this._el_21 = this.renderer.createElement(this._el_17, 'div', null);
        this.renderer.setElementAttribute(this._el_21, 'class', 'md-slider-thumb-label');
        this._text_22 = this.renderer.createText(this._el_21, '\n          ', null);
        this._el_23 = this.renderer.createElement(this._el_21, 'span', null);
        this.renderer.setElementAttribute(this._el_23, 'class', 'md-slider-thumb-label-text');
        this._text_24 = this.renderer.createText(this._el_23, '', null);
        this._text_25 = this.renderer.createText(this._el_21, '\n        ', null);
        this._text_26 = this.renderer.createText(this._el_17, '\n      ', null);
        this._text_27 = this.renderer.createText(this._el_15, '\n    ', null);
        this._text_28 = this.renderer.createText(this._el_2, '\n  ', null);
        this._text_29 = this.renderer.createText(this._el_0, '\n', null);
        this._text_30 = this.renderer.createText(parentRenderNode, '\n', null);
        this._expr_0 = import13.UNINITIALIZED;
        this._expr_1 = import13.UNINITIALIZED;
        this._expr_2 = import13.UNINITIALIZED;
        this._expr_3 = import13.UNINITIALIZED;
        this.init([], [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3,
            this._el_4,
            this._text_5,
            this._el_6,
            this._text_7,
            this._el_8,
            this._text_9,
            this._el_10,
            this._text_11,
            this._el_12,
            this._text_13,
            this._text_14,
            this._el_15,
            this._text_16,
            this._el_17,
            this._text_18,
            this._el_19,
            this._text_20,
            this._el_21,
            this._text_22,
            this._el_23,
            this._text_24,
            this._text_25,
            this._text_26,
            this._text_27,
            this._text_28,
            this._text_29,
            this._text_30
        ], [], []);
        return null;
    };
    _View_MdSlider0.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_0 = this.context.isSliding;
        if (import11.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setElementClass(this._el_2, 'md-slider-sliding', currVal_0);
            this._expr_0 = currVal_0;
        }
        var currVal_1 = this.context.isActive;
        if (import11.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementClass(this._el_2, 'md-slider-active', currVal_1);
            this._expr_1 = currVal_1;
        }
        var currVal_2 = this.context.thumbLabel;
        if (import11.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementClass(this._el_2, 'md-slider-thumb-label-showing', currVal_2);
            this._expr_2 = currVal_2;
        }
        var currVal_3 = import11.interpolate(1, '', this.context.value, '');
        if (import11.checkBinding(throwOnChange, this._expr_3, currVal_3)) {
            this.renderer.setText(this._text_24, currVal_3);
            this._expr_3 = currVal_3;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_MdSlider0;
}(import9.AppView));
export function viewFactory_MdSlider0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_MdSlider === null)) {
        (renderType_MdSlider = viewUtils.createRenderComponentType('', 0, import16.ViewEncapsulation.None, styles_MdSlider, {}));
    }
    return new _View_MdSlider0(viewUtils, parentInjector, declarationEl);
}
