// @angular/material/tabs/tabs.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from '@angular/material/tabs/tabs';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from '@angular/material/core/portal/portal-directives';
import * as import4 from '@angular/common/src/localization';
import * as import6 from '@angular/core/src/i18n/tokens';
import * as import8 from '@angular/core/src/linker/view';
import * as import9 from '@angular/core/src/linker/element';
import * as import10 from '@angular/core/src/linker/query_list';
import * as import11 from '@angular/core/src/linker/view_utils';
import * as import12 from '@angular/core/src/linker/view_type';
import * as import13 from '@angular/core/src/change_detection/change_detection';
import * as import14 from '@angular/core/src/zone/ng_zone';
import * as import15 from '@angular/core/src/metadata/view';
import * as import16 from '@angular/core/src/linker/component_factory';
import * as import17 from './tab-group.css.shim';
import * as import18 from '@angular/common/src/directives/ng_for';
import * as import19 from '@angular/material/tabs/ink-bar';
import * as import20 from '@angular/core/src/linker/template_ref';
import * as import21 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import22 from '@angular/core/src/linker/element_ref';
import * as import23 from '@angular/material/tabs/tab-label-wrapper';
import * as import24 from '@angular/core/src/linker/component_factory_resolver';
import * as import25 from '@angular/common/src/directives/ng_if';
var MdTabsModuleInjector = (function (_super) {
    __extends(MdTabsModuleInjector, _super);
    function MdTabsModuleInjector(parent) {
        _super.call(this, parent, [], []);
    }
    Object.defineProperty(MdTabsModuleInjector.prototype, "_NgLocalization_3", {
        get: function () {
            if ((this.__NgLocalization_3 == null)) {
                (this.__NgLocalization_3 = new import4.NgLocaleLocalization(this.parent.get(import6.LOCALE_ID)));
            }
            return this.__NgLocalization_3;
        },
        enumerable: true,
        configurable: true
    });
    MdTabsModuleInjector.prototype.createInternal = function () {
        this._CommonModule_0 = new import2.CommonModule();
        this._PortalModule_1 = new import3.PortalModule();
        this._MdTabsModule_2 = new import1.MdTabsModule();
        return this._MdTabsModule_2;
    };
    MdTabsModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import2.CommonModule)) {
            return this._CommonModule_0;
        }
        if ((token === import3.PortalModule)) {
            return this._PortalModule_1;
        }
        if ((token === import1.MdTabsModule)) {
            return this._MdTabsModule_2;
        }
        if ((token === import4.NgLocalization)) {
            return this._NgLocalization_3;
        }
        return notFoundResult;
    };
    MdTabsModuleInjector.prototype.destroyInternal = function () {
    };
    return MdTabsModuleInjector;
}(import0.NgModuleInjector));
export var MdTabsModuleNgFactory = new import0.NgModuleFactory(MdTabsModuleInjector, import1.MdTabsModule);
var renderType_MdTabGroup_Host = null;
var _View_MdTabGroup_Host0 = (function (_super) {
    __extends(_View_MdTabGroup_Host0, _super);
    function _View_MdTabGroup_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_MdTabGroup_Host0, renderType_MdTabGroup_Host, import12.ViewType.HOST, viewUtils, parentInjector, declarationEl, import13.ChangeDetectorStatus.CheckAlways);
    }
    _View_MdTabGroup_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('md-tab-group', rootSelector, null);
        this._appEl_0 = new import9.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_MdTabGroup0(this.viewUtils, this.injector(0), this._appEl_0);
        this._MdTabGroup_0_4 = new import1.MdTabGroup(this.parentInjector.get(import14.NgZone));
        this._query_MdTab_0_0 = new import10.QueryList();
        this._appEl_0.initComponent(this._MdTabGroup_0_4, [], compView_0);
        compView_0.create(this._MdTabGroup_0_4, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_MdTabGroup_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import1.MdTabGroup) && (0 === requestNodeIndex))) {
            return this._MdTabGroup_0_4;
        }
        return notFoundResult;
    };
    _View_MdTabGroup_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if (this._query_MdTab_0_0.dirty) {
                this._query_MdTab_0_0.reset([]);
                this._MdTabGroup_0_4._tabs = this._query_MdTab_0_0;
                this._query_MdTab_0_0.notifyOnChanges();
            }
        }
        this.detectViewChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            this._MdTabGroup_0_4.ngAfterViewChecked();
        }
    };
    return _View_MdTabGroup_Host0;
}(import8.AppView));
function viewFactory_MdTabGroup_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_MdTabGroup_Host === null)) {
        (renderType_MdTabGroup_Host = viewUtils.createRenderComponentType('', 0, import15.ViewEncapsulation.None, [], {}));
    }
    return new _View_MdTabGroup_Host0(viewUtils, parentInjector, declarationEl);
}
export var MdTabGroupNgFactory = new import16.ComponentFactory('md-tab-group', viewFactory_MdTabGroup_Host0, import1.MdTabGroup);
var styles_MdTabGroup = [import17.styles];
var renderType_MdTabGroup = null;
var _View_MdTabGroup0 = (function (_super) {
    __extends(_View_MdTabGroup0, _super);
    function _View_MdTabGroup0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_MdTabGroup0, renderType_MdTabGroup, import12.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import13.ChangeDetectorStatus.CheckAlways);
    }
    _View_MdTabGroup0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._viewQuery_MdTabLabelWrapper_0 = new import10.QueryList();
        this._viewQuery_MdInkBar_1 = new import10.QueryList();
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'md-tab-header');
        this.renderer.setElementAttribute(this._el_0, 'role', 'tablist');
        this._text_1 = this.renderer.createText(this._el_0, '\n  ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._appEl_2 = new import9.AppElement(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import20.TemplateRef_(this._appEl_2, viewFactory_MdTabGroup1);
        this._NgFor_2_6 = new import18.NgFor(this._appEl_2.vcRef, this._TemplateRef_2_5, this.parentInjector.get(import21.IterableDiffers), this.ref);
        this._text_3 = this.renderer.createText(this._el_0, '\n  ', null);
        this._el_4 = this.renderer.createElement(this._el_0, 'md-ink-bar', null);
        this._MdInkBar_4_3 = new import19.MdInkBar(this.renderer, new import22.ElementRef(this._el_4));
        this._text_5 = this.renderer.createText(this._el_0, '\n', null);
        this._text_6 = this.renderer.createText(parentRenderNode, '\n', null);
        this._el_7 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_7, 'class', 'md-tab-body-wrapper');
        this._text_8 = this.renderer.createText(this._el_7, '\n  ', null);
        this._anchor_9 = this.renderer.createTemplateAnchor(this._el_7, null);
        this._appEl_9 = new import9.AppElement(9, 7, this, this._anchor_9);
        this._TemplateRef_9_5 = new import20.TemplateRef_(this._appEl_9, viewFactory_MdTabGroup3);
        this._NgFor_9_6 = new import18.NgFor(this._appEl_9.vcRef, this._TemplateRef_9_5, this.parentInjector.get(import21.IterableDiffers), this.ref);
        this._text_10 = this.renderer.createText(this._el_7, '\n', null);
        this._text_11 = this.renderer.createText(parentRenderNode, '\n', null);
        var disposable_0 = this.renderer.listen(this._el_0, 'keydown', this.eventHandler(this._handle_keydown_0_0.bind(this)));
        this._expr_1 = import13.UNINITIALIZED;
        this._expr_2 = import13.UNINITIALIZED;
        this.init([], [
            this._el_0,
            this._text_1,
            this._anchor_2,
            this._text_3,
            this._el_4,
            this._text_5,
            this._text_6,
            this._el_7,
            this._text_8,
            this._anchor_9,
            this._text_10,
            this._text_11
        ], [disposable_0], []);
        return null;
    };
    _View_MdTabGroup0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import20.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import18.NgFor) && (2 === requestNodeIndex))) {
            return this._NgFor_2_6;
        }
        if (((token === import19.MdInkBar) && (4 === requestNodeIndex))) {
            return this._MdInkBar_4_3;
        }
        if (((token === import20.TemplateRef) && (9 === requestNodeIndex))) {
            return this._TemplateRef_9_5;
        }
        if (((token === import18.NgFor) && (9 === requestNodeIndex))) {
            return this._NgFor_9_6;
        }
        return notFoundResult;
    };
    _View_MdTabGroup0.prototype.detectChangesInternal = function (throwOnChange) {
        var changes = null;
        changes = null;
        var currVal_1 = this.context._tabs;
        if (import11.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this._NgFor_2_6.ngForOf = currVal_1;
            if ((changes === null)) {
                (changes = {});
            }
            changes['ngForOf'] = new import13.SimpleChange(this._expr_1, currVal_1);
            this._expr_1 = currVal_1;
        }
        if ((changes !== null)) {
            this._NgFor_2_6.ngOnChanges(changes);
        }
        if (!throwOnChange) {
            this._NgFor_2_6.ngDoCheck();
        }
        changes = null;
        var currVal_2 = this.context._tabs;
        if (import11.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this._NgFor_9_6.ngForOf = currVal_2;
            if ((changes === null)) {
                (changes = {});
            }
            changes['ngForOf'] = new import13.SimpleChange(this._expr_2, currVal_2);
            this._expr_2 = currVal_2;
        }
        if ((changes !== null)) {
            this._NgFor_9_6.ngOnChanges(changes);
        }
        if (!throwOnChange) {
            this._NgFor_9_6.ngDoCheck();
        }
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if (this._viewQuery_MdTabLabelWrapper_0.dirty) {
                this._viewQuery_MdTabLabelWrapper_0.reset([this._appEl_2.mapNestedViews(_View_MdTabGroup1, function (nestedView) {
                        return [nestedView._MdTabLabelWrapper_0_3];
                    })]);
                this.context._labelWrappers = this._viewQuery_MdTabLabelWrapper_0;
                this._viewQuery_MdTabLabelWrapper_0.notifyOnChanges();
            }
            if (this._viewQuery_MdInkBar_1.dirty) {
                this._viewQuery_MdInkBar_1.reset([this._MdInkBar_4_3]);
                this.context._inkBar = this._viewQuery_MdInkBar_1;
                this._viewQuery_MdInkBar_1.notifyOnChanges();
            }
        }
    };
    _View_MdTabGroup0.prototype._handle_keydown_0_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.context.handleKeydown($event) !== false);
        return (true && pd_0);
    };
    return _View_MdTabGroup0;
}(import8.AppView));
export function viewFactory_MdTabGroup0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_MdTabGroup === null)) {
        (renderType_MdTabGroup = viewUtils.createRenderComponentType('', 0, import15.ViewEncapsulation.Emulated, styles_MdTabGroup, {}));
    }
    return new _View_MdTabGroup0(viewUtils, parentInjector, declarationEl);
}
var _View_MdTabGroup1 = (function (_super) {
    __extends(_View_MdTabGroup1, _super);
    function _View_MdTabGroup1(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_MdTabGroup1, renderType_MdTabGroup, import12.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import13.ChangeDetectorStatus.CheckAlways);
    }
    _View_MdTabGroup1.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'md-tab-label');
        this.renderer.setElementAttribute(this._el_0, 'md-tab-label-wrapper', '');
        this.renderer.setElementAttribute(this._el_0, 'role', 'tab');
        this._MdTabLabelWrapper_0_3 = new import23.MdTabLabelWrapper(new import22.ElementRef(this._el_0));
        this._text_1 = this.renderer.createText(this._el_0, '\n    ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._appEl_2 = new import9.AppElement(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import20.TemplateRef_(this._appEl_2, viewFactory_MdTabGroup2);
        this._PortalHostDirective_2_6 = new import3.PortalHostDirective(this.parent.parentInjector.get(import24.ComponentFactoryResolver), this._appEl_2.vcRef);
        this._text_3 = this.renderer.createText(this._el_0, '\n  ', null);
        this._expr_1 = import13.UNINITIALIZED;
        this._expr_2 = import13.UNINITIALIZED;
        this._expr_3 = import13.UNINITIALIZED;
        this._expr_4 = import13.UNINITIALIZED;
        this._expr_5 = import13.UNINITIALIZED;
        this._expr_6 = import13.UNINITIALIZED;
        var disposable_0 = this.renderer.listen(this._el_0, 'click', this.eventHandler(this._handle_click_0_0.bind(this)));
        this._expr_7 = import13.UNINITIALIZED;
        this.init([].concat([this._el_0]), [
            this._el_0,
            this._text_1,
            this._anchor_2,
            this._text_3
        ], [disposable_0], []);
        return null;
    };
    _View_MdTabGroup1.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import20.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import3.PortalHostDirective) && (2 === requestNodeIndex))) {
            return this._PortalHostDirective_2_6;
        }
        if (((token === import23.MdTabLabelWrapper) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) {
            return this._MdTabLabelWrapper_0_3;
        }
        return notFoundResult;
    };
    _View_MdTabGroup1.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_7 = this.context.$implicit.label;
        if (import11.checkBinding(throwOnChange, this._expr_7, currVal_7)) {
            this._PortalHostDirective_2_6.portal = currVal_7;
            this._expr_7 = currVal_7;
        }
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_1 = this.parent.context._getTabLabelId(this.context.index);
        if (import11.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementProperty(this._el_0, 'id', currVal_1);
            this._expr_1 = currVal_1;
        }
        var currVal_2 = ((this.parent.context.selectedIndex == this.context.index) ? 0 : (0 - 1));
        if (import11.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementProperty(this._el_0, 'tabIndex', currVal_2);
            this._expr_2 = currVal_2;
        }
        var currVal_3 = this.parent.context._getTabContentId(this.context.index);
        if (import11.checkBinding(throwOnChange, this._expr_3, currVal_3)) {
            this.renderer.setElementAttribute(this._el_0, 'aria-controls', ((currVal_3 == null) ? null : currVal_3.toString()));
            this._expr_3 = currVal_3;
        }
        var currVal_4 = (this.parent.context.selectedIndex == this.context.index);
        if (import11.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            this.renderer.setElementAttribute(this._el_0, 'aria-selected', ((currVal_4 == null) ? null : currVal_4.toString()));
            this._expr_4 = currVal_4;
        }
        var currVal_5 = (this.parent.context.selectedIndex == this.context.index);
        if (import11.checkBinding(throwOnChange, this._expr_5, currVal_5)) {
            this.renderer.setElementClass(this._el_0, 'md-tab-active', currVal_5);
            this._expr_5 = currVal_5;
        }
        var currVal_6 = this.context.$implicit.disabled;
        if (import11.checkBinding(throwOnChange, this._expr_6, currVal_6)) {
            this.renderer.setElementClass(this._el_0, 'md-tab-disabled', currVal_6);
            this._expr_6 = currVal_6;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_MdTabGroup1.prototype.dirtyParentQueriesInternal = function () {
        this.parent._viewQuery_MdTabLabelWrapper_0.setDirty();
    };
    _View_MdTabGroup1.prototype._handle_click_0_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = ((this.parent.context.focusIndex = (this.parent.context.selectedIndex = this.context.index)) !== false);
        return (true && pd_0);
    };
    return _View_MdTabGroup1;
}(import8.AppView));
function viewFactory_MdTabGroup1(viewUtils, parentInjector, declarationEl) {
    return new _View_MdTabGroup1(viewUtils, parentInjector, declarationEl);
}
var _View_MdTabGroup2 = (function (_super) {
    __extends(_View_MdTabGroup2, _super);
    function _View_MdTabGroup2(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_MdTabGroup2, renderType_MdTabGroup, import12.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import13.ChangeDetectorStatus.CheckAlways);
    }
    _View_MdTabGroup2.prototype.createInternal = function (rootSelector) {
        this.init([], [], [], []);
        return null;
    };
    return _View_MdTabGroup2;
}(import8.AppView));
function viewFactory_MdTabGroup2(viewUtils, parentInjector, declarationEl) {
    return new _View_MdTabGroup2(viewUtils, parentInjector, declarationEl);
}
var _View_MdTabGroup3 = (function (_super) {
    __extends(_View_MdTabGroup3, _super);
    function _View_MdTabGroup3(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_MdTabGroup3, renderType_MdTabGroup, import12.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import13.ChangeDetectorStatus.CheckAlways);
    }
    _View_MdTabGroup3.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'md-tab-body');
        this.renderer.setElementAttribute(this._el_0, 'role', 'tabpanel');
        this._text_1 = this.renderer.createText(this._el_0, '\n    ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._appEl_2 = new import9.AppElement(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import20.TemplateRef_(this._appEl_2, viewFactory_MdTabGroup4);
        this._NgIf_2_6 = new import25.NgIf(this._appEl_2.vcRef, this._TemplateRef_2_5);
        this._text_3 = this.renderer.createText(this._el_0, '\n  ', null);
        this._expr_0 = import13.UNINITIALIZED;
        this._expr_1 = import13.UNINITIALIZED;
        this._expr_2 = import13.UNINITIALIZED;
        this._expr_3 = import13.UNINITIALIZED;
        this.init([].concat([this._el_0]), [
            this._el_0,
            this._text_1,
            this._anchor_2,
            this._text_3
        ], [], []);
        return null;
    };
    _View_MdTabGroup3.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import20.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import25.NgIf) && (2 === requestNodeIndex))) {
            return this._NgIf_2_6;
        }
        return notFoundResult;
    };
    _View_MdTabGroup3.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_3 = (this.parent.context.selectedIndex == this.context.index);
        if (import11.checkBinding(throwOnChange, this._expr_3, currVal_3)) {
            this._NgIf_2_6.ngIf = currVal_3;
            this._expr_3 = currVal_3;
        }
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_0 = this.parent.context._getTabContentId(this.context.index);
        if (import11.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setElementProperty(this._el_0, 'id', currVal_0);
            this._expr_0 = currVal_0;
        }
        var currVal_1 = (this.parent.context.selectedIndex == this.context.index);
        if (import11.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementClass(this._el_0, 'md-tab-active', currVal_1);
            this._expr_1 = currVal_1;
        }
        var currVal_2 = this.parent.context._getTabLabelId(this.context.index);
        if (import11.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementAttribute(this._el_0, 'aria-labelledby', ((currVal_2 == null) ? null : currVal_2.toString()));
            this._expr_2 = currVal_2;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_MdTabGroup3;
}(import8.AppView));
function viewFactory_MdTabGroup3(viewUtils, parentInjector, declarationEl) {
    return new _View_MdTabGroup3(viewUtils, parentInjector, declarationEl);
}
var _View_MdTabGroup4 = (function (_super) {
    __extends(_View_MdTabGroup4, _super);
    function _View_MdTabGroup4(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_MdTabGroup4, renderType_MdTabGroup, import12.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import13.ChangeDetectorStatus.CheckAlways);
    }
    _View_MdTabGroup4.prototype.createInternal = function (rootSelector) {
        this._text_0 = this.renderer.createText(null, '\n      ', null);
        this._anchor_1 = this.renderer.createTemplateAnchor(null, null);
        this._appEl_1 = new import9.AppElement(1, null, this, this._anchor_1);
        this._TemplateRef_1_5 = new import20.TemplateRef_(this._appEl_1, viewFactory_MdTabGroup5);
        this._PortalHostDirective_1_6 = new import3.PortalHostDirective(this.parent.parent.parentInjector.get(import24.ComponentFactoryResolver), this._appEl_1.vcRef);
        this._text_2 = this.renderer.createText(null, '\n    ', null);
        this._expr_0 = import13.UNINITIALIZED;
        this.init([].concat([
            this._text_0,
            this._appEl_1,
            this._text_2
        ]), [
            this._text_0,
            this._anchor_1,
            this._text_2
        ], [], []);
        return null;
    };
    _View_MdTabGroup4.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import20.TemplateRef) && (1 === requestNodeIndex))) {
            return this._TemplateRef_1_5;
        }
        if (((token === import3.PortalHostDirective) && (1 === requestNodeIndex))) {
            return this._PortalHostDirective_1_6;
        }
        return notFoundResult;
    };
    _View_MdTabGroup4.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0 = this.parent.context.$implicit.content;
        if (import11.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this._PortalHostDirective_1_6.portal = currVal_0;
            this._expr_0 = currVal_0;
        }
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_MdTabGroup4;
}(import8.AppView));
function viewFactory_MdTabGroup4(viewUtils, parentInjector, declarationEl) {
    return new _View_MdTabGroup4(viewUtils, parentInjector, declarationEl);
}
var _View_MdTabGroup5 = (function (_super) {
    __extends(_View_MdTabGroup5, _super);
    function _View_MdTabGroup5(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_MdTabGroup5, renderType_MdTabGroup, import12.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import13.ChangeDetectorStatus.CheckAlways);
    }
    _View_MdTabGroup5.prototype.createInternal = function (rootSelector) {
        this.init([], [], [], []);
        return null;
    };
    return _View_MdTabGroup5;
}(import8.AppView));
function viewFactory_MdTabGroup5(viewUtils, parentInjector, declarationEl) {
    return new _View_MdTabGroup5(viewUtils, parentInjector, declarationEl);
}
